scrapy/pipelines/files.py

```diff
@@ -334,6 +336,9 @@ def _warn():
 
         media_guid = hashlib.sha1(to_bytes(url)).hexdigest()  # change to request.url after deprecation
         media_ext = os.path.splitext(url)[1]  # change to request.url after deprecation
+        if not media_ext or not all([char.isalpha() or char.isdigit() for char in media_ext[1:]]):
+            media_base_url = urlparse_cached(request).path
```

Comment:
You must be careful here with [this](https://github.com/scrapy/scrapy/pull/1548/files#diff-0c52d5edf5deca8bf1993b707225b409R327). I know is not part of the PR but that line should transform the string into a request or at least into:
```python
>>> request = type('MockedRequest', (), {'url': 'http://example.com'})
>>> request.url
'http://example.com'
```